src/main/kotlin/com/hibob/academy/service/PetService.kt

```diff
@@ -0,0 +1,42 @@
+package com.hibob.academy.service
+
+import com.hibob.academy.dao.*
+import jakarta.inject.Inject
+import org.springframework.stereotype.Service
+
+@Service
+class PetService @Inject constructor(private val petDao: PetDao) {
+
+    fun createPet(pet: PetCreationRequest): Long = petDao.createPet(pet)
+
+    fun getPetsByType(petType: PetType, companyId: Long) = petDao.getPetsByType(petType, companyId)
+
+    fun getPetById(petId: Long, companyId: Long): PetData? = petDao.getPetById(petId, companyId)
+
+    fun getAllPets(companyId: Long) = petDao.getAllPets(companyId)
+
+    fun updatePetOwner(petId: Long, ownerId: Long, companyId: Long) {
+        val success = petDao.updatePetOwner(petId, ownerId, companyId) > 0
+        if(!success)
+            throw IllegalStateException("Pet doesn't exist or pet already has an owner")
+    }
+
+    fun updatePetName(petId: Long, newName: String, companyId: Long) = petDao.updatePetName(petId, newName, companyId)
+
+    fun deletePet(petId: Long, companyId: Long) = petDao.deletePet(petId, companyId)
+
+    fun getPetsByOwner(ownerId: Long, companyId: Long) = petDao.getPetsByOwner(ownerId, companyId)
+
+    fun countPetsByType(companyId: Long) = petDao.countPetsByType(companyId)
+
+    fun adoptPets(ownerId: Long, companyId: Long, petsIds: List<Long>): Map<Long, Boolean> {
+        return petsIds.associate { petId ->
```

Comment:
BTW - there is a better performance way to update all at once
set owner where pet id in